tests/test_remote_sequential.py

```diff
@@ -39,10 +40,10 @@ def test_remote_sequential():
     assert hidden.shape == test_inputs.shape
     assert hidden.requires_grad
     second_half_outputs = second_half(hidden)
-    assert torch.allclose(second_half_outputs, full_outputs)
+    assert torch.allclose(second_half_outputs, full_outputs, atol=1e-4)
```

Comment:
related to TP: TP partial matrix products are added up in non-deterministic order. The 1e-4 tolerance is chosen because (1) errors accumulate for 24 layers and (2) we need some leeway to make sure tests aint flappy.